pandas/_libs/reduction.pyx

```diff
@@ -459,14 +463,25 @@ cdef class BlockSlider:
         self.index._engine.clear_mapping()
         self.index._cache.clear()  # e.g. inferred_freq must go
 
+    cdef _restore_blocks(self):
```

Comment:
Is it possible to wrap this in a context manager instead of doing it this way?

Reply:
we do it at each call of `move`, so i dont think its like a try/finally